File: java/CreateReturn.java

import com.alibaba.fastjson.JSONException;
import com.alibaba.fastjson.JSONObject;

import java.io.IOException;
import java.text.ParseException;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class CreateReturn {
	
	public static void main(String[] args) throws JSONException, IOException, ParseException, FodelAPIException{
		String secretKey = "";//secretKey from Fodel
		String requestTime = (System.currentTimeMillis()/1000)+"";
		Map<String ,String> params = new HashMap<String,String>();
		params.put("recipient_name", "Alex Meng");//name of the customer
		params.put("phone", "971522545710");//phone number of the customer
		params.put("order_no", "23324234");//order number
		params.put("address", "Dubai Opera ,Dubai");//address of customer
		params.put("app_key", "");//app key from fodel
		params.put("weight", "1200");//weight of the parcels, unit:g
		params.put("return_type", "1");
		//return type,
		//1,warranty(the customer is want to refund),
		//2,return (the customer want to change the product or the product has problem, need to send back the warehouse),
		//3,RTO (return to original)
		params.put("shop_id", "2"); //shop_id ,from the map of the
		params.put("ts", requestTime);//request time

		List<ReturnGoodList> returnGoodLists = new ArrayList<>();
		//product information
		//
		returnGoodLists.add(new ReturnGoodList(Byte.parseByte("2"),"Iphone 7","123456","12.5","12 x 12 x 34"));
		returnGoodLists.add(new ReturnGoodList(Byte.parseByte("3"),"T shirt nike","1213123","12.5","12 x 12 x 34"));
		params.put("product_info",JSONObject.toJSONString(returnGoodLists));
		String signString = EncryptUtil.generateSign(params, secretKey);
		params.put("sign", signString);
		JSONObject jsonObject = HttpRequestUtil.request("POST", "/parcel/return", params);
		System.out.println(jsonObject);
		
	}
   
}

File: php/CreateReturn.php
<?php

/**
 * create an return waybill
 */
include_once "EncryptUtil.php";
include_once "HttpRequestUtil.php";

$productInfo = [
    [   "name"=>"Iphone 7 golden color",//name of the product
        "serial_number"=>"123456",//serial number of the product,
        "qty"=>1, //quantity of the product
        "price"=>2599, //price of the product
        "volume"=>"23 x 12 x 12" //shop id of the customer choose,Length x Width x Height
    ],
    [
        "name"=>"T shirt nike",//name of the product
        "serial_number"=>"123243",//serial number of the product,
        "qty"=>1, //quantity of the product
        "price"=>23,//price of the product
        "volume"=>"23 x 12 x 12" //shop id of the customer choose,Length x Width x Height
    ]
];

$requestTime = time(); //timestamp of request
$secretKey = ""; //secretKey from Fodel
$params = array();
$params["recipient_name"] = "Gloria Davies";//full name of customer
$params["phone"] = "0528144434";//phone of the customer
$params["address"] = "Dubai Opera ,Dubai,United Arab Emirates";//address of the customer
$params["app_key"] = "";//app key of the fodel
//return type :
//1,warranty(the customer is want to refund),
//2,return (the customer want to change the product or the product has problem, need to send back the warehouse),
//3,RTO (return to original)
$params["return_type"] = 1;
$params["weight"] = 1200;
$params["order_no"] = 232342233;//order number of the customer order
$params["shop_id"] = 36;//shop id of the customer choose
$params["product_info"] = json_encode($productInfo); //product information , just put the json format
$params["ts"] = $requestTime;
//generate the sign from the paramters with eht secretKey
$signString = EncryptUtil::generateSign($params,$secretKey);
$params['sign'] = $signString;

//Set header ,add the api version to the parameter fodel_api_version
$header = array();
$header[] = "fodel_api_version: v1";
$httprequest = new HttpRequestUtil();
$httprequest->setBaseUrl("http://api.test.fo-del.com");
$respone = $httprequest->post("/shipment/return",json_encode($params),true,$header);
echo $respone->body;



